editoast/src/modelsv2/work_schedules.rs

```diff
@@ -0,0 +1,53 @@
+use crate::schema::TrackRange;
+use chrono::NaiveDateTime;
+use editoast_derive::ModelV2;
+use strum_macros::FromRepr;
+
+use crate::error::Result;
+use crate::modelsv2::Changeset;
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+#[derive(Debug, Default, Clone, Copy, Serialize, Deserialize, FromRepr, ToSchema, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum WorkScheduleType {
+    #[default]
+    Catenary,
+    Track,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, ModelV2, ToSchema)]
+#[model(table = crate::tables::work_schedule)]
+pub struct WorkSchedule {
+    pub id: i64,
+    pub start_date_time: NaiveDateTime,
+    pub end_date_time: NaiveDateTime,
+    #[model(json)]
+    pub track_ranges: Vec<TrackRange>,
+    pub obj_id: String,
+    #[model(to_enum)]
+    pub work_schedule_type: WorkScheduleType,
+}
+
+fn dates_in_order(a: Option<NaiveDateTime>, b: Option<NaiveDateTime>) -> bool {
```

Comment:
It's usually a weird pattern to take `Option` has input argument (either you call the function with arguments, or the `Option` is empty and you don't call the function at all).

Maybe you can write something like the following instead?

```rust
work_schedule_changeset.start_date_time
  .zip(work_schedule_changeset.end_date_time)
  .map(|a, b| a <= b)
  .unwrap_or(true);
```

Reply:
Thanks for your comment, I removed this function anyway and moved the check in a custom deserializer :)